### DISCUSSION TITLE
How to detect if the application is running under the development server?

### DISCUSSION THREAD

**Discussion by fluca1978** [mojolicious/mojo#2158] [Q&A] [answered]
Is there a way to understand if the application is running under the development server (e.g., `scripts/myapp daemon` or `morbo`)?
I would like to configure differently the application depending on the runtime environment.

**Comment by Grinnz** [comment]
Often the best solution for this is to test `$app->mode`, which is set by default to 'development', or 'production' when running hypnotoad, but can be set as works best for your situation (just note that the default error pages in 'development' mode are quite verbose so ensure that you always deploy in 'production' mode to users).

**Comment by guest20** [comment, marked as answer]
mojo will load config files based on the mode, so it makes sense to put your features behind a config flag and then turn them on in `$appname.development.conf`.

This also lets you turn on a features you planned to be dev-only in non-dev when they turn out to be handy for tracking down issues in prod.

**Comment by jberger** [comment]
I think you can use the [`before_server_start` hook](https://docs.mojolicious.org/Mojolicious#before_server_start)

```bash
$ perl -Mojo -E 'app->hook(before_server_start => sub ($server, $app) { say ref $server }); app->start' prefork
Mojo::Server::Prefork
Web application available at http://127.0.0.1:3000
```

**Reply by jberger** [reply]
Although it doesn't look like that mechanism can distinguish between daemon and morbo, so perhaps not
